app/models/activity.rb

```diff
@@ -277,6 +277,24 @@ def own_and_descendants_transactions
     Transaction.where(parent_activity_id: activity_ids)
   end
 
+  def reportable_transactions_for_level
+    reportable_transactions = if programme?
+      own_and_descendants_transactions
+    else
+      transactions
+    end.order("date DESC")
+
+    return reportable_transactions unless programme?
+
+    spend_by_financial_quarter(reportable_transactions)
+  end
+
+  private def spend_by_financial_quarter(reportable_transactions)
+    reportable_transactions.group_by(&:own_financial_quarter).map do |financial_quarter, actuals|
+      Transaction.new(date: financial_quarter.end_date, value: actuals.sum(&:value), transaction_type: Transaction::DEFAULT_TRANSACTION_TYPE)
```

Comment:
Could we add a description to the Transaction to explain what activity the transaction is for?

Reply:
Yeah, we discussed this with Meyric when we were planning out how to tackle it. We weren't sure what to use as a description for the grouped activities/child activities by financial quarter. The description isn't required so we decided to leave it off.

Reply:
Ah, fair enough 👍